### PR TITLE
AMCL Readme

### PR DESCRIPTION
**Pull request by mhpanah**
First draft for AMCL readme file.

### CODE REVIEW

File: nav2_amcl/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# AMCL
+Adaptive Monte Carlo Localization (AMCL) is a probabilistic localization module which estimates the position and orientation (i.e. Pose) of a robot in a given known map.
+
+## Overview
+Currently, the AMCL module in ROS 2 Navigation System is a direct port from ROS1 AMCL package (http://wiki.ros.org/amcl) with some minor code re-factoring. The direct port includes all of ROS1 functionalities except dynamic parameter reconfiguration and running from Bag files. The code re-factoring includes moving Sensor, Particle Filter, and Map from nav2_amcl directory to nav2_util directory. The main logic behind this is code re-usability.  However, given the way in which these classes are currently written, these codes are highly bounded to AMCL.  Therefore, to make use of code re-usability, these classes need to be substantially modified.
```
Comment (mkhansen-intel) [inline nav2_amcl/README.md]: I would split this paragraph up to the what and why

File: nav2_amcl/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+## Current Plan
+* Polishing AMCL core code, specially the `laserReceived` callback
```
Comment (mkhansen-intel) [inline nav2_amcl/README.md]: Please file issues and provide links to any issues here for planned work, so readers can click for more info on these items.

**Comment by mikeferguson** [comment]
One note about AMCL -- I know costmap was initially ported from the kinetic-devel branch -- if AMCL wasn't ported from melodic-devel, please look at the changes that we made to get to the release -- specifically, we found that the parameters did NOT match the book/description, and fixed that in Melodic. We should definitely make sure that makes it into nav2.

**Comment by mkhansen-intel** [comment]
@mikeferguson - thanks for the heads-up on that, we'll verify today, if those changes aren't here let's open an issue to track and make sure they get in.

File: nav2_amcl/README.md (hunk 2 above)
Comment (mhpanah) [reply nav2_amcl/README.md]: Done.

**Pull request merged**